### DISCUSSION TITLE
Facing issue with selecting an item from sap.ui.core.ListItem element

### DISCUSSION THREAD

**Discussion by Vikrant87-lego** [ui5-community/wdi5#542] [Q&A]
Hi All,

I am trying to select an item from sap.ui.core.ListItem control type using the visibletext but the control does not have any method to select item by visible text.

<img width="427" alt="image" src="https://github.com/ui5-community/wdi5/assets/85674150/bb4fbd10-c68b-4ab6-9021-76f84197b423">

i tried to find the way to get it with UI5 as below:

this.byId("planbookfilter").getItems().find(p => p.getProperty("text").startsWith("Vikrant")).getId()

Is there a way to select the coreitem matching the visibletext just like shown above?

Regards,
Vikrant

**Comment by vobu** [comment]
yes, there is - UI5's select list implementation only renders the list items after the dropdown was opened.
so you'll need to first get the control ([don't forget `interaction: 'root'`](https://ui5-community.github.io/wdi5/#/recipes?id=using-interaction-on-a-selector)), then `$wdi5Control.open()` it - after that, the select list aggregation is filled and accessible. 
See the full recipe over at https://ui5-community.github.io/wdi5/#/recipes?id=test-a-sapmcombobox-or-sapmmulticombobox

**Reply by Vikrant87-lego** [reply]
I tried your solution but i am getting below mentioned error:
[0-0] TypeError in "Test.Should select planbook"
TypeError: selectlist.open is not a function
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[0-0] FAILED in chrome - /webapp/test/testcontrol.test.js

code:
 const selector1= {
            selector: {
                controlType: "sap.m.SelectList",
                id: "__list0",
                searchOpenDialogs: true,
				interaction: "root"
            }
        
        }
        const selectlist= await browser.asControl(selector1)
        await selectlist.open()


        const aItems=await browser.asControl(selector1).getItems(true)

**Comment by vobu** [comment]
please check the syntax of your code, it should read
```js
const selectlist = await browser.asControl(selector1)
await selectlist.open()
const aItems = await selectlist.getItems(true)
```

**Reply by Vikrant87-lego** [reply]
Still getting the same error:

"TypeError: selectlist.open is not a function
at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[0-0] FAILED in chrome - /webapp/test/testcontrol.test.js"

**Comment by vobu** [comment]
then your selector might not be correct and not retrieve the select list you're after.
have you tried [debugging](https://ui5-community.github.io/wdi5/#/debugging) the step and see what you get as `selectlist`?
